File: src/main/java/com/project/food_delivery/dtos/OrderDto.java
package com.project.food_delivery.dtos;

import com.project.food_delivery.models.Payment;
import com.project.food_delivery.models.Status;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class OrderDto {
    private LocalDateTime dateAndTime;
    private Double price;
    private Payment payment;
    private Status status;
    private List<ProductMetadataDto> productMetadata;
}

File: front-end/src/app/common/order-dto.ts
import {Product} from "./product";

export class OrderDto {
    public dateAndTime: Date;
    public price : number;
    public payment : string;
    public status : string;
    public productMetadata : Product[];

    constructor(dateAndTime: Date, price: number, payment: string, status: string, productMetadata: Product[]) {
        this.dateAndTime = dateAndTime;
        this.price = price;
        this.payment = payment;
        this.status = status;
        this.productMetadata = productMetadata;
    }
}
